web/src/input_pill.ts

```diff
@@ -83,28 +83,32 @@ type InputPillRenderingDetails = {
 };
 
 // These are the functions that are exposed to other modules.
-export type InputPillContainer<T> = {
+export type InputPillContainer<ItemType> = {
```

Comment:
Given how these are used, maybe `PillType` would be a more natural name?

```
$ git grep 'InputPill.*<'
web/src/email_pill.ts:export type EmailPillWidget = InputPillContainer<EmailPill>;
web/src/email_pill.ts:    pill_container: input_pill.InputPillContainer<EmailPill>,
web/src/email_pill.ts:): input_pill.InputPillContainer<EmailPill> {
web/src/input_pill.ts:type InputPillCreateOptions<ItemType> = {
web/src/input_pill.ts:type InputPill<ItemType> = {
web/src/input_pill.ts:type InputPillStore<ItemType> = {
web/src/input_pill.ts:    pills: InputPill<ItemType>[];
web/src/input_pill.ts:    pill_config: InputPillCreateOptions<ItemType>["pill_config"];
web/src/input_pill.ts:    create_item_from_text: InputPillCreateOptions<ItemType>["create_item_from_text"];
web/src/input_pill.ts:    get_text_from_item: InputPillCreateOptions<ItemType>["get_text_from_item"];
web/src/input_pill.ts:    get_display_value_from_item: InputPillCreateOptions<ItemType>["get_display_value_from_item"];
web/src/input_pill.ts:    generate_pill_html: InputPillCreateOptions<ItemType>["generate_pill_html"];
web/src/input_pill.ts:    onPillRemove?: (pill: InputPill<ItemType>, trigger: RemovePillTrigger) => void;
web/src/input_pill.ts:export type InputPillContainer<ItemType> = {
web/src/input_pill.ts:    getByElement: (element: HTMLElement) => InputPill<ItemType> | undefined;
web/src/input_pill.ts:        callback: (pill: InputPill<ItemType>, trigger: RemovePillTrigger) => void,
web/src/input_pill.ts:    _get_pills_for_testing: () => InputPill<ItemType>[];
web/src/input_pill.ts:    opts: InputPillCreateOptions<ItemType>,
web/src/input_pill.ts:): InputPillContainer<ItemType> {
web/src/input_pill.ts:    const store: InputPillStore<ItemType> = {
web/src/input_pill.ts:            const payload: InputPill<ItemType> = {
web/src/input_pill.ts:    const prototype: InputPillContainer<ItemType> = {
web/src/search_pill.ts:export type SearchPillWidget = InputPillContainer<SearchPill>;
web/src/stream_pill.ts:export type StreamPillWidget = InputPillContainer<StreamPill>;
web/src/typeahead_helper.ts:export type CombinedPillContainer = InputPillContainer<CombinedPill>;
web/src/user_group_pill.ts:type UserGroupPillWidget = InputPillContainer<UserGroupPill>;
web/src/user_pill.ts:export type UserPillWidget = InputPillContainer<UserPill>;
web/src/user_pill.ts:): input_pill.InputPillContainer<UserPill> {
```

Reply:
I think we can revisit this after further refactoring rounds; seems better to merge this checkpoint than rework that, given how much is stacked on top of that name.

Reply:
When I removed `InputPillItem` in a later commit, you get at the end things like `items: () => ItemType[];` which is why I named it that way